PYIC-5953: Clean up session credential table switch

Wynndow opened the pull request:
**Not to be merged until https://github.com/govuk-one-login/ipv-core-common-infra/pull/906, and we're happy with the changes running in prod**

<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Clean up session credential table switch

### Why did it change

This should be merged once the reads from the sessions credenentials table are happening in prod, and we're happy. It does a few things:

* Removes use of the sessionCredentialsTableReads feature flag
* Stops dual writing VCs. This stops writing to the user-issued-credentials-table during a journey. It's now only written to at the end of a journey
* Removes now unneccesary AWS perms from some lambdas
* Removes some now unncessary filtering from the "working" set of VCs. We should be operating on VCs of a know profile type when creating shared attributes and building identities


### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-5953](https://govukverify.atlassian.net/browse/PYIC-5953)


[PYIC-5954]: https://govukverify.atlassian.net/browse/PYIC-5954?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[PYIC-5953]: https://govukverify.atlassian.net/browse/PYIC-5953?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Wynndow committed:
PYIC-5954: Clean up session credential table switch

This should be merged once the reads from the sessions credenentials
table are happening in prod, and we're happy. It does a few things:

* Removes use of the sessionCredentialsTableReads feature flag
* Removes use of the sessionCredentialsTableWrites feature flag
to at the end of a journey
* Removes now unneccesary AWS perms from some lambdas
* Removes some now unncessary filtering from the "working" set of VCs.
  We should be operating on VCs of a know profile type when creating
shared attributes and building identities

Joe-Edwards-GDS reviewed (commented):
Looks reasonable, but won't approve until we're ready to merge.

Do you think we should also remove the feature flags from `CoreFeatureFlag.java`?

Wynndow committed:
PYIC-5954: Remove sessionCredentialsTableRead flag from journey maps

This removes the now unneeded flag from the journey maps. It's assumed
that when TICF is in play we'll want to go eval-gpg-45 -> ticf -> store.
This is in case we received a breaching CI from TICF and the user ends
up not proving their identity.

Wynndow committed:
PYIC-5954: Stop storing VCs on session item

We were previously storing all VC recevied on the users session item. We
were only using this as a flag for if a user was in an inherited
identity migration journey.

We can replace the list of VC with a flag for if they're in a migration
journey.

Joe-Edwards-GDS commented on libs/common-services/src/main/java/uk/gov/di/ipv/core/library/persistence/item/IpvSessionItem.java:
```diff
@@ -41,7 +39,7 @@ public class IpvSessionItem implements DynamodbItem {
     // Not for general use.
     private String featureSet;
-    private List<String> vcReceivedThisSession;
+    private boolean inheritedIdentityReceivedThisSession;
```
I think this is fine for now - we might need to rethink this entirely once we introduce EVCS though, see notes on https://govukverify.atlassian.net/browse/PYIC-6083

Joe-Edwards-GDS replied:
(It's also worth checking this won't break any existing sessions, but I think it should be fine...)

Wynndow replied:
I figured that as we're not live with inherited identities this should be fine. The reads and writes to this are with behind a flag, or only read if the user has an operation profile.
